subsys/power/device.c

```diff
@@ -22,7 +22,8 @@ LOG_MODULE_DECLARE(power);
  * to build the device list based on devices power
  * and clock domain dependencies.
  */
-#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF51X)
+#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF51X) \
```

Comment:
Can we change it to for example ```SOC_FAMILY_NRF``` ?
This line contains all supported Nordic's MCU families

Reply:
Apparently, regulator peripheral is available only from Application core in nRF5340.
So this could be either
`#if defined(SOC_FAMILY_NRF) && !defined(CONFIG_SOC_NRF5340_CPUNET)`
or 
`
#if defined(CONFIG_SOC_SERIES_NRF52X) || defined(CONFIG_SOC_SERIES_NRF51X) \
    || defined(CONFIG_SOC_SERIES_NRF5340_CPUAPP) || defined(CONFIG_SOC_SERIES_NRF91X)
`

Also, I think that `select HAS_SYS_POWER_STATE_DEEP_SLEEP_1` should be moved from `SOC_SERIES_NRF53X` to `SOC_SERIES_NRF5340_CPUAPP`.

Reply:
> Apparently, regulator peripheral is available only from Application core in nRF5340.
> So this could be either
> `#if defined(SOC_FAMILY_NRF) && !defined(CONFIG_SOC_NRF5340_CPUNET)`

But this part of code is related to the device (not system) power management. The REGULATOR peripheral is not involved here, so there is no need to exclude the network core.

> Also, I think that `select HAS_SYS_POWER_STATE_DEEP_SLEEP_1` should be moved from `SOC_SERIES_NRF53X` to `SOC_SERIES_NRF5340_CPUAPP`.

I agree.

Reply:
Changed to `#if defined(SOC_FAMILY_NRF)` and moved `select HAS_SYS_POWER_STATE_DEEP_SLEEP_1`.